crates/scroll/alloy/provider/src/engine/provider.rs

```diff
@@ -0,0 +1,135 @@
+    use reth_scroll_node::ScrollEngineValidator;
+    use reth_scroll_payload::NoopPayloadJobGenerator;
+    use reth_tasks::TokioTaskExecutor;
+    use reth_transaction_pool::noop::NoopTransactionPool;
+    use std::net::{Ipv4Addr, SocketAddr, SocketAddrV4};
+    use tokio::sync::mpsc::unbounded_channel;
+
+    fn spawn_test_payload_service<T>() -> PayloadBuilderHandle<T>
+    where
+        T: PayloadTypes<
+                PayloadBuilderAttributes = ScrollPayloadBuilderAttributes,
+                BuiltPayload = ScrollBuiltPayload,
+            > + 'static,
+    {
+        let (service, handle) = PayloadBuilderService::<
+            NoopPayloadJobGenerator<ScrollPayloadBuilderAttributes, ScrollBuiltPayload>,
+            futures_util::stream::Empty<CanonStateNotification>,
+            T,
+        >::new(Default::default(), futures_util::stream::empty());
+        tokio::spawn(service);
+        handle
+    }
+
+    async fn launch_auth(jwt_secret: JwtSecret) -> AuthServerHandle {
+        let config = AuthServerConfig::builder(jwt_secret)
+            .socket_addr(SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::UNSPECIFIED, 0)))
+            .build();
+        let (tx, _rx) = unbounded_channel();
+        let beacon_engine_handle = BeaconConsensusEngineHandle::<ScrollEngineTypes>::new(tx);
+        let client = ClientVersionV1 {
+            code: ClientCode::RH,
+            name: "Reth".to_string(),
+            version: "v0.2.0-beta.5".to_string(),
+            commit: "defa64b2".to_string(),
+        };
+
+        let engine_api = EngineApi::new(
+            NoopProvider::default(),
+            SCROLL_MAINNET.clone(),
+            beacon_engine_handle,
+            spawn_test_payload_service().into(),
+            NoopTransactionPool::default(),
+            Box::<TokioTaskExecutor>::default(),
+            client,
+            EngineCapabilities::default(),
+            ScrollEngineValidator::new(SCROLL_MAINNET.clone()),
+        );
+        let module = AuthRpcModule::new(engine_api);
+        module.start_server(config).await.unwrap()
+    }
+
+    #[allow(unused_must_use)]
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_engine_api_provider() -> eyre::Result<()> {
+        reth_tracing::init_test_tracing();
+
+        let secret = JwtSecret::random();
+        let handle = launch_auth(secret).await;
+        let url = handle.http_url().parse()?;
+        let provider = ScrollAuthEngineApiProvider::new(secret, url);
+
+        let block = Block::<TransactionSigned>::default().seal_slow();
+        provider.new_payload_v1(block_to_payload_v1(block.clone())).await;
+        provider.fork_choice_updated_v1(ForkchoiceState::default(), None).await;
+        provider.get_payload_v1(PayloadId::new([0, 0, 0, 0, 0, 0, 0, 0])).await;
+        provider.get_payload_bodies_by_hash_v1(vec![]).await;
+        provider.get_payload_bodies_by_range_v1(U64::ZERO, U64::from(1u64)).await;
+        provider.exchange_capabilities(vec![]).await;
```

Comment:
this comes from [here](https://github.com/scroll-tech/reth/blob/83b2fb9b413b7386d77414e573a8874fb9d83151/crates/rpc/rpc-builder/tests/it/auth.rs#L18), and doesn't use the `?` operator because there is no functioning `BeaconConsensus` running in the test.